Update assemblies and modules Docs for Metrics Reporter integration (#11329)
strimzi/strimzi-kafka-operator: documentation/modules/metrics/proc-strimzi-metrics-reporter-kafka-deploy-options.adoc

// This assembly is included in the following assemblies:
//
// metrics/assembly_metrics-kafka.adoc

[id='proc-metrics-reporter-kafka-deploy-options-{context}']

> = Enabling Strimzi Metrics Reporter metrics

Reviewer: do we need metrics on the end or can we match with the JMX Exporter title: "Enabling Strimzi Metrics Reporter "

// This assembly is included in the following assemblies:
//
// metrics/assembly_metrics-kafka.adoc

[id='proc-metrics-reporter-kafka-deploy-options-{context}']

> = Enabling Strimzi Metrics Reporter metrics through configuration

Reviewer: ```suggestion
= Enabling Strimzi Metrics Reporter
```

[role="_abstract"]
To enable and expose metrics in Strimzi for Prometheus, use metrics configuration properties.

The following components require `metricsConfig` configuration to expose metrics in Prometheus format:

* Kafka

This configuration enables the {StrimziMetricsReporter} to expose metrics through an HTTP endpoint on port 9404.
The port for the Strimzi Metrics Reporter HTTP endpoint is 9404.
Prometheus scrapes this endpoint to collect Kafka metrics.

> To deploy Prometheus metrics configuration in Strimzi, you can use your own configuration or the xref:ref-metrics-reporter-config-{context}[example custom resource configuration file] provided with Strimzi:

Reviewer: ```suggestion
You can create your own Prometheus configuration or use the xref:ref-metrics-prometheus-metrics-config-{context}[example custom resource file] provided with Strimzi:
```

* `kafka-metrics.yaml`

> This file contain the necessary configuration to enable Prometheus metrics.
> It is a good starting point for trying Prometheus with Strimzi.

Reviewer: ```suggestion
This file contains the necessary configuration, and is a good starting point for trying Prometheus with Strimzi.
```

> This procedure shows how to deploy example Prometheus metrics configuration in the `Kafka` resource.

Reviewer: ```suggestion
This procedure shows how to deploy example Prometheus metrics configuration to the `Kafka` resource.
```

> The process is the same when deploying the example files for other resources.

Reviewer: ```suggestion

```

Reviewer: we include a note/cross-ref on how to configure the Kafka Exporter as well in the JMX procedure. Do we want it here too?

Lines 28-31:  https://github.com/strimzi/strimzi-kafka-operator/pull/11307/files#diff-80eda6a9cce800533455d246c9b923ba06d53ea83645996177b63b5ef4f33e66R28-R31

Or maybe that needs to be separated from both now, so we link from each of these procedures. Note taken

// This assembly is included in the following assemblies:
//
// metrics/assembly_metrics-kafka.adoc

[id='proc-metrics-reporter-kafka-deploy-options-{context}']

> = Enabling Metrics Reporter metrics through configuration

Reviewer 2: ```suggestion
= Enabling Strimzi Metrics Reporter through configuration
```

[role="_abstract"]

> To enable and expose metrics in Strimzi for Prometheus, use metrics configuration properties.
> 
> The following components require `metricsConfig` configuration to expose metrics:

Reviewer 2: ```suggestion
The following components require `metricsConfig` configuration to expose metrics in Prometheus format:
```

* Kafka

> This configuration enables the {StrimziMetricsReporter} to expose metrics through an HTTP endpoint.
> The port for the Strimzi Metrics Reporter HTTP endpoint is 9404.

Reviewer 2: ```suggestion
This configuration enables the {StrimziMetricsReporter} to expose metrics through an HTTP endpoint on port 9404.
```

and similar change for the Prometheus JMX Exporter paragraph.

* `kafka-metrics.yaml`

This file contain the necessary configuration to enable Prometheus metrics.
It is a good starting point for trying Prometheus with Strimzi.

This procedure shows how to deploy example Prometheus metrics configuration in the `Kafka` resource.
The process is the same when deploying the example files for other resources.

.Procedure

. Deploy the example custom resource with the Prometheus configuration.
+
For example, for each `Kafka` resource you can apply the `kafka-metrics.yaml` file.
+
.Deploying the example configuration
[source,shell,subs="+attributes"]
----
kubectl apply -f kafka-metrics.yaml
----
+
Alternatively, copy the example configuration in `kafka-metrics.yaml` to your own `Kafka` resource.
+
.Copying the example configuration
[source,shell]
----
kubectl edit kafka <kafka_configuration_file>
----
+
Copy the `metricsConfig` property to your `Kafka` resource.
+
.Example metrics configuration for Kafka
[source,yaml,subs="+quotes,attributes"]
----
apiVersion: {KafkaApiVersion}
kind: Kafka
metadata:
  name: my-cluster
spec:
  kafka:
    # ...

>     metricsConfig: <1>

Reviewer 2: There is no reference for <1>. Is it really needed?

      type: strimziMetricsReporter
      values:
        allowList:
          - "kafka_log.*"
          - "kafka_network.*"

>   # _metrics configuration..._

Reviewer 2: Do we need this?
